### DISCUSSION TITLE
WSA Sideloader turns 2 - Future plans

### DISCUSSION THREAD

**Discussion by infinitepower18** [infinitepower18/WSA-Sideloader#108] [Announcements]
Hi everyone, I have some news to share regarding the WSA Sideloader application.

### WSA Sideloader turns 2

WSA Sideloader will soon be released two years ago. This was my first project I released to the public and I'm really happy to see it being used by many people around the world. The app now has over 200,000 downloads across Microsoft Store, GitHub and winget. Thank you to everyone who downloaded, used and reviewed the app on the Microsoft Store. WSA Sideloader continues to be one of the popular options for installing APK files thanks to its simplicity and reliability.

When I first started this project, I decided to use Python as that was one of the languages that I knew that time. I had also used PySimpleGUI to speed up the process of creating the UI which had helped focus on the stability of the app. However as time went on, I became limited by the technical constraints of what this UI framework could provide, such as being able to provide a true Fluent Design look. I had also received feedback that the app is not well accessible to those with special needs, which unfortunately is limited by the capabilities of this UI framework. #92

### Transition to a new UI framework

Today, PySimpleGUI had announced that it will be transitioning to a **paid subscription model** for those who wish to distribute apps. While the framework continues to be free for those experimenting with it, the free option is not viable for applications like WSA Sideloader, as doing so would require every user to use their own license key. While the previous 4.x version remains free to use, it does also mean I cannot use any new features that may be coming in the future without having to pay a yearly subscription, which may not be viable for some open source applications. Since I was planning to change the UI framework anyway, partly because of accessibility requirements, this has accelerated my plans.

**WSA Sideloader will soon be rewritten to use a new UI framework.** While I haven't made a final decision on which framework to use yet, it will continue to be Python based so I can continue to use the same code used for installing APKs which has been proven to work quite well. Possible options include PyQt6, PySide6 or wxPython. Please do state your thoughts on which framework to go for, especially those with accessibility needs. I'm also hoping to bring a more Fluent design like experience with the framework transition.

I don't have a specific release date for the rewritten version yet, but I plan to release it sometime in 2024. I might also launch a beta program while developing the new version, so keep an eye on this GitHub for updates. In the meantime, I will not be adding any new features to the program, only providing bug fixes if necessary. At the moment I am currently maintaining 3 apps, WSA Sideloader, [WSA System Control](https://github.com/infinitepower18/WSA-SystemControl) and [SubManager](https://getsubmanager.app). SubManager is currently my focus and I will need to set priorities accordingly. But the rewrite of WSA Sideloader is on my priority list.

### Commitment to open-source

I do also want to assure you that **WSA Sideloader will continue to be free and open-source**. It has been from the beginning and I have no plans to change this, including with this rewritten version. If you would like to support me financially, I have a Kofi donation page up which you can access [here](https://ko-fi.com/F1F1K06VY). Alternatively, if you use any Apple device, you can download my SubManager app from the [App Store](https://apps.apple.com/us/app/submanager-manage-subs/id1632853914) and purchase the SubManager+ option. Your support is greatly appreciated and will enable me to continue producing high quality apps.

Thank you for taking the time to read this, and I look forward to releasing the new version soon. 🙂

**Comment by MesterPerfect** [comment]
If your primary concern is accessibility support, the optimal choice would be wxPython. However, keep in mind that you might miss out on aesthetically pleasing designs, dark mode, and new features since the wxPython framework doesn't seamlessly provide these.

If your goal is to prioritize accessibility while maintaining good aesthetics, your best options are PyQt6 or PySide6. Follow the accessibility guidelines provided by either of them and test early versions of the software with users who rely on accessibility support.

Feel free to reach out if you need assistance in this regard. 😊 I wish you success and look forward to seeing new releases soon.

**Reply by infinitepower18** [reply]
Yeah, I'm leaning more towards these options. Thanks for the input 🙂
